vault/resource_generic_secret.go

```diff
@@ -121,13 +137,23 @@ func genericSecretResourceWrite(d *schema.ResourceData, meta interface{}) error
 	}
 
 	log.Printf("[DEBUG] Writing generic Vault secret to %s", path)
-	_, err = client.Logical().Write(path, data)
+	secret, err := client.Logical().Write(path, data)
 	if err != nil {
 		return fmt.Errorf("error writing to Vault: %s", err)
 	}
 
 	d.SetId(originalPath)
 
+	if secret != nil {
+		jsonData, err := json.Marshal(secret.Data)
+		if err != nil {
+			return fmt.Errorf("error marshaling JSON for %q: %s", path, err)
+		}
+		d.Set("write_data_json", string(jsonData))
```

Comment:
Writing the entire secret out gives me pause, though I'm undecided what I think of it yet.

Reply:
Yeah, it makes me a bit nervous too, actually. I wonder whether there should be an input parameter that specifically indicates which fields should be written.